local/compose/create.go

```diff
@@ -296,6 +296,32 @@ func buildContainerMountOptions(p types.Project, s types.ServiceConfig, inherit
 		}
 		mounts = append(mounts, mount)
 	}
+	for _, secret := range s.Secrets {
+		if contains(inherited, secret.Target) {
+			continue
+		}
+		target := "/var/run/secrets"
+		if secret.Target == "" {
+			target = fmt.Sprintf("%s/%s", target, secret.Source)
+		} else if !filepath.IsAbs(secret.Target) {
+			target = fmt.Sprintf("%s/%s", target, secret.Target)
+		}
+		defSecret, ok := p.Secrets[secret.Source]
+
+		if !ok {
```

Comment:
this is already checked by compose-go loader
https://github.com/compose-spec/compose-go/blob/master/loader/validate.go#L51